src/common/contentPreference.ts

```diff
@@ -309,16 +324,24 @@ const blockSource: BlockEntity = async ({ ctx, id }) => {
       sourceId: id,
       feedId: ctx.userId,
       status: ContentPreferenceStatus.Blocked,
+      referralToken: randomUUID(),
     });
 
-    await repository.save(contentPreference);
+    await repository
+      .createQueryBuilder()
+      .insert()
+      .into(ContentPreferenceSource)
+      .values(contentPreference)
+      .orUpdate(['status'], ['referenceId', 'userId'])
+      .execute();
```

Comment:
Changed this to upsert so that `referralToken` is preserved when moving from `follow` -> `subscribed`